providers/cncf/kubernetes/src/airflow/providers/cncf/kubernetes/operators/job.py

```diff
@@ -168,34 +170,43 @@ def execute(self, context: Context):
         ti.xcom_push(key="job_namespace", value=self.job.metadata.namespace)
 
         self.pod: k8s.V1Pod | None
```

Comment:
this property seems to be internal to the KubernetesJobOperator itself. Is that the case? If so does it make sense to remove it and only keep "self.pods"? And also unify the cases where parallelism is None or 1 with parallelism > 1?